config/routes.rb

```diff
@@ -41,6 +44,12 @@
     post '/webhook', to: 'webhook#message'
   end
 
+  namespace :whats_app do
+    post '/webhook', to: 'webhook#message'
+    post '/errors', to: 'webhook#errors'
+    post '/status', to: 'webhook#status'
```

Comment:
After talking to @mattwr18 I understand that Twilio sends read state etc. on this webhook.

Reply:
Yes, see [Track Delivery Status of Messages in Ruby](https://www.twilio.com/docs/sms/tutorials/how-to-confirm-delivery-ruby).

At the moment, we are just using it to track `UNSUCCESSFUL_DELIVERY` and raising an error. I consider it "unsuccessful" if the read status is `undelivered` or `failed`. I saw this when trying to send a freeform message to a contributor who is unable to receive one because the have not sent us a message in more than 24 hours.

It would also happen if there is some difference between the approved templates in our WhatsApp page and what we have in the application. For example, a typo, or someone edits the WhatsApp template, but we don't update the application.

I noticed that we don't get an error if we try to send a freeform message when one is not permitted, the message just never gets delivered. I configured this so that we actually do raised an error and we can track them and fix them.